core/crypto/primitives/aes.go

```diff
@@ -39,108 +38,101 @@ func GenAESKey() ([]byte, error) {
-	ciphertext := make([]byte, aes.BlockSize+len(s))
-	iv := ciphertext[:aes.BlockSize]
-	if _, err := io.ReadFull(rand.Reader, iv); err != nil {
-		return nil, err
+	blockSize := aes.BlockSize + len(message)
+	tmpCiphertext := make([]byte, blockSize)
+	randVector := tmpCiphertext[:aes.BlockSize]
+	_, error = io.ReadFull(rand.Reader, randVector)
+
+	if error != nil {
+		return nil, error
 	}
 
-	mode := cipher.NewCBCEncrypter(block, iv)
-	mode.CryptBlocks(ciphertext[aes.BlockSize:], s)
+	encMode := cipher.NewCBCEncrypter(cipherBlock, randVector)
+	encMode.CryptBlocks(tmpCiphertext[aes.BlockSize:], message)
 
-	// It's important to remember that ciphertexts must be authenticated
-	// (i.e. by using crypto/hmac) as well as being encrypted in order to
-	// be secure.
-	return ciphertext, nil
+	return tmpCiphertext, nil
 }
 
-// CBCDecrypt decrypts using CBC mode
-func CBCDecrypt(key, src []byte) ([]byte, error) {
-	block, err := aes.NewCipher(key)
+// AESDecryptCBC decrypts a AES block with CBC mode
+func AESDecryptCBC(decKey, cipherText []byte) ([]byte, error) {
+	decBlock, err := aes.NewCipher(decKey)
+
 	if err != nil {
 		return nil, err
 	}
 
-	// The IV needs to be unique, but not secure. Therefore it's common to
-	// include it at the beginning of the ciphertext.
-	if len(src) < aes.BlockSize {
-		return nil, errors.New("ciphertext too short")
+	if len(cipherText) < aes.BlockSize {
+		return nil, errors.New("ciphertext needs to be multiple of 16")
 	}
-	iv := src[:aes.BlockSize]
-	src = src[aes.BlockSize:]
 
-	// CBC mode always works in whole blocks.
-	if len(src)%aes.BlockSize != 0 {
+	initVector := cipherText[:aes.BlockSize]
+	originalMsg := cipherText[aes.BlockSize:]
+
+	if len(cipherText)%aes.BlockSize != 0 {
 		return nil, errors.New("ciphertext is not a multiple of the block size")
 	}
 
-	mode := cipher.NewCBCDecrypter(block, iv)
-
-	// CryptBlocks can work in-place if the two arguments are the same.
-	mode.CryptBlocks(src, src)
-
-	// If the original plaintext lengths are not a multiple of the block
-	// size, padding would have to be added when encrypting, which would be
-	// removed at this point. For an example, see
-	// https://tools.ietf.org/html/rfc5246#section-6.2.3.2. However, it's
-	// critical to note that ciphertexts must be authenticated (i.e. by
-	// using crypto/hmac) before being decrypted in order to avoid creating
-	// a padding oracle.
+	decMode := cipher.NewCBCDecrypter(decBlock, initVector)
+	decMode.CryptBlocks(originalMsg, cipherText[aes.BlockSize:])
 
-	return src, nil
+	return PKCS7UnPadding(originalMsg)
 }
 
 // CBCPKCS7Encrypt combines CBC encryption and PKCS7 padding
 func CBCPKCS7Encrypt(key, src []byte) ([]byte, error) {
-	return CBCEncrypt(key, PKCS7Padding(src))
+	return AESEncryptCBC(key, PKCS7Padding(src))
 }
 
 // CBCPKCS7Decrypt combines CBC decryption and PKCS7 unpadding
 func CBCPKCS7Decrypt(key, src []byte) ([]byte, error) {
-	pt, err := CBCDecrypt(key, src)
+	pt, err := AESDecryptCBC(key, src)
```

Comment:
+1 for adding the AES (despite working on `aes.go`. Golang does not really care much about filenames, eh?).

Since we use `AES` in `CBC` mode (and then have Encrypt/Decrypt), how about we go for: `AESCBCEncrypt` ?